src/app/qgsbookmarks.h

```diff
@@ -18,10 +18,87 @@
 #define QGSBOOKMARKS_H
 
 #include <QDockWidget>
+#include <QSqlTableModel>
 
 #include "ui_qgsbookmarksbase.h"
 #include "qgscontexthelp.h"
 
+
+/*
+ * Model that merge the QGIS and project model
+ */
+class QgsBookmarksProjectTableModel: public QAbstractTableModel
+{
+    Q_OBJECT
+
+  public:
+
+    QgsBookmarksProjectTableModel();
+
+    int rowCount( const QModelIndex& parent = QModelIndex() ) const override;
+
+    int columnCount( const QModelIndex& parent = QModelIndex() ) const override;
+
+    QVariant data( const QModelIndex& index, int role = Qt::DisplayRole ) const override;
+
+    bool setData( const QModelIndex& index, const QVariant& value, int role = Qt::EditRole ) override;
+
+    bool insertRows( int row, int count, const QModelIndex& parent = QModelIndex() ) override;
+
+    bool removeRows( int row, int count, const QModelIndex& parent = QModelIndex() ) override;
+
+  private slots:
+    void projectRead() { emit layoutChanged(); };
+};
+
+/*
+ * Model that merge the QGIS and project model
+ */
+class QgsBookmarksMergeTableModel: public QAbstractTableModel
```

Comment:
These two classes have the same comments.
I am not a native english speaker, but I would find better to name them QgsProjectBookmarksTableModel and QgsMergedBookmarksTableModel. What do you think ?